library/ssl_cli.c

```diff
@@ -2936,6 +2988,28 @@ static int ssl_write_client_key_exchange( mbedtls_ssl_context *ssl )
     }
     else
 #endif /* MBEDTLS_KEY_EXCHANGE_DHE_RSA_ENABLED */
+#if defined(MBEDTLS_USE_UECC)
+    if (ciphersuite_info->key_exchange == MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA)
+    {
+        uint8_t own_pubkey[2*NUM_ECC_BYTES];
+        uint8_t own_privatekey[NUM_ECC_BYTES];
+
+        const struct uECC_Curve_t * curve = uECC_secp256r1();
+
+        //TODO: provide rng for uECC
```

Comment:
Leftover todo (again, just to have a comment that can be 'resolved' once this has been removed)